Makefile

```diff
@@ -204,14 +205,14 @@ helm-docs: $(HELM_DOCS) ## Use helm-docs to generate documentation from helm cha
 		-t ../../../Documentation/Helm-Charts/ceph-cluster-chart.gotmpl.md \
 		-t ../../../Documentation/Helm-Charts/_templates.gotmpl
 
-check.helm-docs:
-	@$(MAKE) helm-docs
+check.helm-docs: gen.helm-docs
+	@tests/scripts/validate_modified_files.sh helm-docs
+
 	@git diff --exit-code || { \
 	echo "Please run 'make helm-docs' locally, commit the updated docs, and push the change. See https://rook.io/docs/rook/latest/Contributing/documentation/#making-docs" ; \
 	exit 2 ; \
 	};
-check.docs:
-	@$(MAKE) docs
+check.docs: gen.docs
 	@tests/scripts/validate_modified_files.sh docs
```

Comment:
We had a discussion last month where Travis and I agreed that local checks that use `git` don't add value for end users. For users, `check.docs` is really doing the same thing that `gen.docs` does and then checks `git`. However, a dev could have any number of modifications, so the git check will give frequent false negatives (positives?), and potentially misleading runaround for newer devs.

That said, I would prefer to remove these `check.docs` and `check.helm-docs` targets and replace GH actions usage with direct usage of the `tests/scripts/validate_modified_files.sh` script.

We can encourage users to use `gen.` targets locally, specifically `make generate` if they are unsure of what to run. That will allow them to check `git` themselves, which they should already be doing. Our CI will accurately detect missing files with specifics that aren't misleading.

Reply:
done